Spigot-Server-Patches/0492-Add-villager-reputation-API.patch

```diff
@@ -0,0 +1,125 @@
++import java.util.Map; // Paper
++import java.util.UUID; // Paper
+ import net.minecraft.server.EntityVillager;
+ import net.minecraft.server.IRegistry;
+ import net.minecraft.server.VillagerProfession;
+@@ -112,4 +117,58 @@ public class CraftVillager extends CraftAbstractVillager implements Villager {
+     public static VillagerProfession bukkitToNmsProfession(Profession bukkit) {
+         return IRegistry.VILLAGER_PROFESSION.get(CraftNamespacedKey.toMinecraft(bukkit.getKey()));
+     }
++
++    // Paper start - Add villager reputation API
++    @Override
++    public Reputation getReputation(UUID uniqueId) {
++        net.minecraft.server.Reputation.a rep = getHandle().getReputation().getReputations().get(uniqueId);
++        if (rep == null) {
++            return new Reputation(Maps.newHashMap());
++        }
++
++        Map<ReputationType, Integer> reputationMap = Maps.newHashMap();
++        reputationMap.put(ReputationType.MAJOR_NEGATIVE, rep.getReputationValues().getOrDefault(net.minecraft.server.ReputationType.MAJOR_NEGATIVE, 0));
++        reputationMap.put(ReputationType.MAJOR_POSITIVE, rep.getReputationValues().getOrDefault(net.minecraft.server.ReputationType.MAJOR_POSITIVE, 0));
++        reputationMap.put(ReputationType.MINOR_NEGATIVE, rep.getReputationValues().getOrDefault(net.minecraft.server.ReputationType.MINOR_NEGATIVE, 0));
++        reputationMap.put(ReputationType.MINOR_POSITIVE, rep.getReputationValues().getOrDefault(net.minecraft.server.ReputationType.MINOR_POSITIVE, 0));
++        reputationMap.put(ReputationType.TRADING, rep.getReputationValues().getOrDefault(net.minecraft.server.ReputationType.TRADING, 0));
++        return new Reputation(reputationMap);
++    }
++
++    @Override
++    public Map<UUID, Reputation> getReputations() {
++        return getHandle().getReputation().getReputations().entrySet()
++            .stream()
++            .collect(java.util.stream.Collectors.toMap(Map.Entry::getKey, value -> {
++                Map<ReputationType, Integer> reputationMap = Maps.newHashMap();
++                reputationMap.put(ReputationType.MAJOR_NEGATIVE, value.getValue().getReputationValues().getOrDefault(net.minecraft.server.ReputationType.MAJOR_NEGATIVE, 0));
++                reputationMap.put(ReputationType.MAJOR_POSITIVE, value.getValue().getReputationValues().getOrDefault(net.minecraft.server.ReputationType.MAJOR_POSITIVE, 0));
++                reputationMap.put(ReputationType.MINOR_NEGATIVE, value.getValue().getReputationValues().getOrDefault(net.minecraft.server.ReputationType.MINOR_NEGATIVE, 0));
++                reputationMap.put(ReputationType.MINOR_POSITIVE, value.getValue().getReputationValues().getOrDefault(net.minecraft.server.ReputationType.MINOR_POSITIVE, 0));
++                reputationMap.put(ReputationType.TRADING, value.getValue().getReputationValues().getOrDefault(net.minecraft.server.ReputationType.TRADING, 0));
++                return new Reputation(reputationMap);
```

Comment:
Could we reuse code instead of doing code duplication for these few lines? Doesn't really matter, but it triggers me a bit. Well, you know... I'm triggery.

Reply:
Should be fixed now.